### ISSUE TITLE
Theia can incorrectly generate "Unsupported activation event" error in stdout

### ISSUE THREAD

**Issue by jcortell68** [eclipse-theia/theia#12953]
### Bug Description:
A Theia product that introduces new plugin API can also introduce new [activation events](https://code.visualstudio.com/api/references/activation-events)

Introducing a new activation is quite straightforward. You really just call `HostedPluginSupport.activateByEvent(activationEvent: string)` from within your Theia extension to effectively "generate the event". Theia's implementation of that function simply activates any plugins that have that string declared in their package.json as an `activationEvent`. So really, it's just a meeting of the mind between your product-specific Theia extension and the developers writing plugins for your product. If you two are on the same page, then it all just works., You don't need your Theia extension to somehow register the new activation event with the plugin mechanism. The activation event is ultimately just an opaque string to the plugin mechanism.

But...Theia tries to warn folks installing VS Code extensions when an extension declares an `activationEvent` that the Theia platform itself doesn't support (doesn't generate). And to do that, it has to have knowledge of what activation events it supports. That static list is harcoded into Theia [here](https://github.com/eclipse-theia/theia/blob/990908be18b5045280d2fa12c2be12dc3c7bdb82/packages/plugin-ext/src/plugin/plugin-manager.ts#L82)

Problem is, it doesn't consider that a product developer might be introducing new activation events. And so while it will infact support any and all foreign activation events, it will end up warning the user via stdout that it _doesn't_ support them. And that's obviously not right. We need some way for a Theia extension to tell the aforementioned code: "hey, here are some additional activation event that I support", so that it doesn't generate that error msg in stdout.

### Steps to Reproduce:
```
git clone -b activation_event_gap https://github.com/jcortell68/theiasandbox
cd theiasandbox
yarn
yarn start:electron
```
Note the error message in stdout
```
root ERROR [hosted-plugin: <pid>] Unsupported activation events: onSayHello, please open an issue: https://github.com/eclipse-theia/theia/issues/new
```
Also note that the activation event works quite well despite that misguided message :-)

**Comment by jcortell68** [comment]
@jfaltermeier I took a quick look at the commit. Doesn't look like you started a pull request on it, so not sure if you're looking for feedback or not. Not sure if that's the fix you're planning on going with, but if you do, could you please describe how you envision the environment variable approach being used in practice. I.e., how do you foresee the environment variable being added by a deployed application? I looked for other cases of environment variables being used to alter/extend runtime Theia app behavior and didn't find any (not in the Theia codebase, anyway).

BTW, I'm on vacation then next ten days, but I'll be checking in occasionally.

**Comment by jfaltermeier** [comment]
I have opened a PR by now to gather some feedback. 
I am not 100% what is the best way to specify the events that should not get logged. I went for the env variable as an initial suggestion, because it allows modification for an already built application as well. So users that install additional extension and now get those warnings, may tweak their environment accordingly. 

In a browser application, which I expect to get containerized in some way, the env variable may be set e.g. in the Dockerfile. 
For an electron application I would suggest a custom `main` script that hands over to the main theie script, e.g.:
```
process.env.ADDITIONAL_ACTIVATION_EVENTS = 'onProfile,onOpenExternalUri,onWalkthrough';
// Handover to the auto-generated electron application handler.
require('../lib/backend/electron-main.js');
```
